internal/model/puller.go

```diff
@@ -341,12 +353,33 @@ func (p *Puller) pullerIteration(ignores *ignore.Matcher) int {
 		return true
 	})
 
+nextFile:
 	for {
 		fileName, ok := p.queue.Pop()
 		if !ok {
 			break
 		}
 		if f, ok := p.model.CurrentGlobalFile(p.folder, fileName); ok {
+			if !f.IsSymlink() {
+				key := string(f.Blocks[0].Hash)
+				for i, candidate := range buckets[key] {
+					if scanner.BlocksEqual(candidate.Blocks, f.Blocks) {
+						// Remove the candidate from the bucket
+						l := len(buckets[key]) - 1
+						buckets[key][i] = buckets[key][l]
```

Comment:
No, l in that case would be 0, then we'd do:
_[0] = _[0]
_ = _[:0]
making _ an empty slice.

The error is at line 339, which is file.Blocks[0], somehow the file which comes out of the database has no blocks.